### PR TITLE
Test Suites: UsersController and PostsController

### PR DESCRIPTION
**Pull request by stino-x**
Description
This pull request introduces thorough test suites for both the UsersController and PostsController in our Rails application. The tests cover various actions, ensuring the functionality and correctness of the controller methods.

Changes Made
UsersController
Added tests for the index action to check for a successful response, rendering the correct template, and assigning all users to @users.
Included tests for the new action to verify a successful response and rendering of the new template.
Implemented tests for the create action, covering scenarios with both valid and invalid parameters.
Introduced tests for the show action, checking for a successful response, rendering the show template, and assigning the requested user to @user.
Added tests for the edit action, ensuring a successful response, rendering the edit template, and assigning the requested user to @user.
Included tests for the update action, covering the updating of user attributes and redirection to the updated user.
Implemented tests for the destroy action, checking the destruction of a user and redirection to the users index.
PostsController
Added tests for the index action to check for a successful response, rendering the correct template, and assigning all posts to @posts.
Included tests for the show action, ensuring a successful response, rendering the show template, and assigning the requested post to @post.
Introduced tests for creating a new post in the create action, covering scenarios with both valid and invalid parameters.
Added tests for the edit action, ensuring a successful response, rendering the edit template, and assigning the requested post to @post.
Included tests for the update action, covering the updating of post attributes and redirection to the updated post.
Implemented tests for the destroy action, checking the destruction of a post and redirection to the posts index.
Quality Assurance
All tests have been run locally and pass successfully.
Code adheres to the established coding standards.
The test coverage provides confidence in the correctness of the UsersController and PostsController functionalities.
Additional Notes
Please review the changes, and if approved, I'll merge the pull request to ensure the reliability of our controllers.

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  describe 'GET #index' do
+    it 'returns a successful response' do
+      user = User.create(name: 'John Doe')
```
Comment (V-Blaze) [inline spec/controllers/posts_controller_spec.rb]: - You have done great so far :+1: :+1: But there are a few requirements you still need to meet before moving to the next milestone :+1: :small_airplane: 

- - Screenshot reference:
Selection_038

- [ ] As stated in the project requirement, after checking that the ` response status was correct.` and also that `a correct template was rendered.` You need to also check `If the response body includes the correct placeholder text. ` Currently, you do not have a check for that. At this current milestone, your view files don't need to contain any code logic, just a placeholder text. eg: the post index could have `<h2>Here is a list of all the posts from this user</h2>`. :smile:  Then there should be a test checking if the correct place holder was rendered. eg:
```ruby
    it 'Includes the correct placeholder text' do
      expect(response.body).to include('Here is a list of all the posts from this user')
    end
```

This is what the requirement says and you are expected to do this on the `Post index and Post show`. I'll suggest you implement this correctly. :smile: :waffle:

File: spec/controllers/user_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+RSpec.describe UsersController, type: :controller do
+  describe 'GET #index' do
+    it 'returns a successful response' do
+      get :index
+      expect(response).to be_successful
+    end
+
+    it 'renders the index template' do
+      get :index
```
Comment (V-Blaze) [inline spec/controllers/user_controller_spec.rb]: - [ ] Just as I pointed out above, You are meant to check if the correct placeholder text was rendered, you need to add placeholder text in the `users index and show` view file and then follow the direction I suggested above to see if the body includes the placeholder. I'll suggest you implement these accordingly. :smile: :wave:

File: spec/controllers/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
```
Comment (V-Blaze) [inline spec/controllers/posts_controller_spec.rb]: - [ ] For these tests, you are actually testing the requests and not the controller actions. THe Rspec type is meant to be request `type: request`. I'll suggest you update this correctly. `RSpec.describe 'Posts', type: :request do`.

- [ ] Also these tests are meant to be in the `requests` folder in your spec file and not in the `controllers` folder. I'll suggest you update these and move these tests to the correct folder location. :smile: :small_airplane:

File: spec/controllers/user_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# spec/controllers/users_controller_spec.rb
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
```
Comment (V-Blaze) [inline spec/controllers/user_controller_spec.rb]: - [ ] Just as I suggested above, these tests are of `type` request and not controller, I'll suggest you update the types to match the correct type of test cases here. eg: `RSpec.describe 'Users', type: :request do` :smile: :small_airplane:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @stino-x 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Current tests are passing ✔
- Clean and well written codes👋 
-  Correct git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Austin,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on implementing recommended changes!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Let's make the ReadMe stand out even better by including instructions under the _Getting Started_ section, on how to run test cases for the app, for instance, using the command `rspec`.

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**